src/simtools/production_configuration/scale_events_manager.py

```diff
@@ -0,0 +1,117 @@
+events and performing interpolation. It is designed to handle the evaluation of statistical
+uncertainties from DL2 Monte Carlo (MC) event files and perform interpolation for a specified
+query point.
+
+The module includes functionality to:
+- Initialize evaluators for statistical error calculations based on input parameters.
+- Perform interpolation using the initialized evaluators to estimate scaled events at a query point.
+- Write the results of the interpolation to an output file.
+
+"""
+
+import itertools
+import json
+import logging
+from pathlib import Path
+
+import astropy.units as u
+import numpy as np
+
+from simtools.io_operations import io_handler
+from simtools.production_configuration.calculate_statistical_errors_grid_point import (
+    StatisticalErrorEvaluator,
+)
+from simtools.production_configuration.interpolation_handler import InterpolationHandler
+from simtools.utils.general import collect_data_from_file
+
+
+class ScaleEventsManager:
+    """
+    Manages the workflow for scaling events and performing interpolation.
+
+    This class handles the evaluation of statistical uncertainties from DL2 MC event files
+    and performs interpolation for a specified query point.
+    """
+
+    def __init__(self, args_dict):
+        """
+        Initialize the manager with the provided arguments.
+
+        Parameters
+        ----------
+        args_dict : dict
+            Dictionary of command-line arguments.
+        """
+        self.args = args_dict
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logging.INFO)
+        self.output_path = io_handler.IOHandler().get_output_directory(Path(__file__).stem)
+        self.output_filepath = Path(self.output_path).joinpath(f"{self.args['output_file']}")
+        self.metrics = collect_data_from_file(self.args["metrics_file"])
+        self.evaluator_instances = []
+
+    def initialize_evaluators(self):
+        """Initialize StatisticalErrorEvaluator instances for the given zeniths and offsets."""
+        if not (self.args["base_path"] and self.args["zeniths"] and self.args["offsets"]):
+            self.logger.warning("No files read")
+            self.logger.warning(f"Base Path: {self.args['base_path']}")
+            self.logger.warning(f"Zeniths: {self.args['zeniths']}")
+            self.logger.warning(f"Offsets: {self.args['offsets']}")
+            return
+
+        for zenith, offset in itertools.product(self.args["zeniths"], self.args["offsets"]):
+            file_name = self.args["file_name_template"].format(zenith=int(zenith))
+            file_path = Path(self.args["base_path"]).joinpath(file_name)
+
+            if not file_path.exists():
+                self.logger.warning(f"File not found: {file_path}. Skipping.")
+                continue
+
+            evaluator = StatisticalErrorEvaluator(
+                file_path,
+                file_type="Gamma-cone",
```

Comment:
Fixed file type? What if I would look at protons?

Reply:
I removed the file_type now, as this was not used anywhere in the code and I am not sure this is needed if we just want to derive the events for a certain production.